app/services/lms_sync_service.py

```diff
@@ -86,6 +99,191 @@ async def sync_assignments(self):
                 )
         
         return canvas_assignments
+    
+    async def _pair_db_canvas_users(self, db_users: list[UserModel], canvas_users: list[dict], canvas_user_pids: list[str]):
+        # User exists in Canvas but not DB = create
+        # User exists in DB but not Canvas = delete
+        # User exists in both DB and Canvas = update
+        pairs = []
+        
+        async def process_canvas_user(canvas_user) -> ResourcePair | None:
+            pid, email, name = canvas_user.get("sis_user_id"), canvas_user.get("email"), canvas_user.get("name")
+            if pid is None or email is None or name is None:
+                print("Skipping over pending user", name or email or "<unknown>")
+                return None
+
+            try:
+                print("getting user info for ", pid)
+                user_info = self.ldap_service.get_user_info(pid)
+                print(pid, "->", user_info.onyen)
+            except:
```

Comment:
I think this should probably be an 
```
except UserNotFoundException:
    print("Skipping over student not in LDAP: ", pid or email or "<unknown>")
except:
    print("Some other error occurred")
```
kind of thing because currently you're swallowing all kinds of exceptions that the LDAP search could throw. If we run into a problem later where LDAP is failing consistently but we're seeing "Skipping over student not in LDAP:" in the logs and can't figure out why for a while that would suck.

Reply:
yes, that's true. I only copied the existing logic, but I can change it.